torchvision/prototype/models/resnet.py

```diff
@@ -304,26 +305,32 @@ class Wide_ResNet101_2_Weights(WeightsEnum):
     DEFAULT = IMAGENET1K_V2
 
 
-@handle_legacy_interface(weights=("pretrained", ResNet18_Weights.IMAGENET1K_V1))
-def resnet18(*, weights: Optional[ResNet18_Weights] = None, progress: bool = True, **kwargs: Any) -> ResNet:
-    """ResNet-18 from `Deep Residual Learning for Image Recognition <https://arxiv.org/pdf/1512.03385.pdf>`__.
+def _make_resnet_docstring(weights):
+    name = weights.__name__
+    num_layers = re.search("ResNet(.+)_Weights", name).groups()[0]
+    allowed_values = ", ".join(f"``{v}``" for v in weights)
+
+    return f"""ResNet-{num_layers} from `Deep Residual Learning for Image Recognition <https://arxiv.org/pdf/1512.03385.pdf>`__.
 
     Args:
-        weights (:class:`~torchvision.prototype.models.ResNet18_Weights`, optional): The
-            pretrained weights to use. See
-            :class:`~torchvision.prototype.models.ResNet18_Weights` below for
-            more details, and possible values. By default, no pre-trained
-            weights are used.
+        weights (:class:`~torchvision.prototype.models.{name}`, optional): The
+            pretrained weights to use. Possible values are {allowed_values}.
```

Comment:
Note that printing the possible allowed values as we do here is only possible with the proposed changes, not with the current implementation (red part on the right). I mean technically we could, but we would have to list them manually and we would have to update the docstring every time we add new weights, so it's not robust.